src/ray/gcs/gcs_server/gcs_health_check_manager.cc

```diff
@@ -82,15 +82,45 @@ std::vector<NodeID> GcsHealthCheckManager::GetAllNodes() const {
   return nodes;
 }
 
+void GcsHealthCheckManager::MarkNodeHealthy(const NodeID &node_id) {
+  io_service_.dispatch(
+      [this, node_id]() {
+        thread_checker_.IsOnSameThread();
```

Comment:
need RAY_CHECK ?

Reply:
can you also add [[nodiscard]] to the checker method definition

Reply:
makes sense! fixed.